src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -51,6 +53,7 @@ object DotnetRunner extends Logging {
     // In debug mode this runner will not launch a .NET process.
     val runInDebugMode = settings._1
     @volatile var dotnetBackendPortNumber = settings._2
+    val dotnetBackendIPAddress = settings._3
```

Comment:
This will be a breaking change.  Also in the PR description you say this is optional.  How are you differentiating this vs the csharp.exe ?  Maybe set this via env variable ?

Reply:
I was parsing the value to see if it can parse to an IP address, but making it a configurable environment variable makes more sense, I'll do that

Reply:
I don't think the `InetAddress.getByName(..)` would have been enough to guard against something like a csharp application coincidentally named after some hostname.